starter-code/basic-algorithms.js

```diff
@@ -5,3 +5,65 @@
 
 
 // Lorem ipsum generator
+
+var hacker1= "Jana";
+var hacker2= prompt("what is the navigator's name?");
+hacker2 = hacker2.toLowerCase();
+console.log("The driver's name is " + hacker1);
+console.log("The navigator's name is " + hacker2);
+
+if (hacker1.length > hacker2.length) {
+  console.log("The Driver has the longest name, it has " + hacker1.length + " characters");
+}
+else if (hacker1.length < hacker2.length) {
+  console.log("Yo, navigator got the longest name, it has " + hacker2.length + " characters");
+}
+else if (hacker1.length === hacker2.length) {
+  console.log("wow, you both got equally long names, " + hacker1.length + " characters");
+}
+
+var str="";
+for (var i = 0; i < hacker1.length; i++) {
+  console.log(hacker1[i].toUpperCase());
+  str += hacker1[i].toUpperCase() + " ";
+}
+console.log(str);
+
+console.log(hacker2.split('').reverse().join(''));
+
+if(hacker1.toLowerCase().split('').sort() < hacker2.split('').sort()){
```

Comment:
It won't work like this, as you are sorting all the letters, per exemple in Jana, as there is two "a" it will be sorted before "arthur" (only one "a") but it shouldn't ;) 

You only have to remove the .sort() method and you code will be good !

```js
var hacker1Lower = hacker1.toLowerCase();
var hacker2Lower = hacker2.toLowerCase();

if (hacker1Lower > hacker2Lower) {
console.log("The navigator's name goes first");
} else if (hacker1Lower < hacker2Lower) {
  console.log("The driver's name goes first");
} else {
  console.log("What?! You both got the same name?");
};

```